core/server/master/src/main/java/alluxio/master/block/DefaultBlockMaster.java

```diff
@@ -689,6 +696,54 @@ public BlockInfo getBlockInfo(long blockId) throws BlockInfoException, Unavailab
     return ret;
   }
 
+  private MasterWorkerInfo lookupWorker(WorkerNetAddress workerNetAddress) {
+    for (IndexedSet<MasterWorkerInfo> workers: Arrays.asList(mTempWorkers, mLostWorkers)) {
+      MasterWorkerInfo worker = workers.getFirstByField(ADDRESS_INDEX, workerNetAddress);
+      if (worker != null) {
+        return worker;
+      }
+    }
+    return null;
+  }
+
+  private MasterWorkerInfo lookupWorker(long workerId) {
+    for (IndexedSet<MasterWorkerInfo> workers: Arrays.asList(mTempWorkers, mLostWorkers)) {
+      MasterWorkerInfo worker = workers.getFirstByField(ID_INDEX, workerId);
+      if (worker != null) {
+        return worker;
+      }
+    }
+    return null;
+  }
+
+  private MasterWorkerInfo moveWorker(long workerId) {
+    for (IndexedSet<MasterWorkerInfo> workers: Arrays.asList(mTempWorkers, mLostWorkers)) {
+      MasterWorkerInfo worker = workers.getFirstByField(ID_INDEX, workerId);
+      if (worker == null) {
+        continue;
+      }
+
+      synchronized (worker) {
+        worker.updateLastUpdatedTimeMs();
+        mWorkers.add(worker);
+        workers.remove(worker);
+        if (workers ==  mLostWorkers) {
+          for (Consumer<Address> function : mLostWorkerFoundListeners) {
+            function.accept(new Address(worker.getWorkerAddress().getHost(),
+                  worker.getWorkerAddress().getRpcPort()));
+          }
+          LOG.warn("A lost worker {} has requested its old id {}.",
+              worker.getWorkerAddress(), worker.getId());
+        } else {
+          LOG.info("=== move worker {}:{}", worker.getId(), worker.getWorkerAddress());
```

Comment:
I don't think we need to log this, since there is another log message in `workerRegister()`.

Reply:
that works

Reply:
removed